### DISCUSSION TITLE
Configure the project locally to enable debugging and use breakpoints

### DISCUSSION THREAD

**Discussion by wjimenez123** [infiniflow/ragflow#3275] [General]
I would like to know if it's possible to configure the project locally to enable debugging and use breakpoints. I am currently trying to do this in Visual Studio Code; however, although the project runs correctly, the breakpoints are not working.

For the backend, I used a .sh file with the following content:

#!/bin/bash

#! Activate the virtual environment
source .venv/bin/activate

#! Export PYTHONPATH
export PYTHONPATH=$(pwd)

#! Run the backend startup script
bash docker/launch_backend_service.sh
For the frontend, the configuration in the launch.json file is as follows:

{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Debug Backend (launch_backend_service.sh)",
            "type": "node",
            "request": "launch",
            "runtimeExecutable": "bash",
            "args": ["${workspaceFolder}/docker/start_backend_debug.sh"],
            "console": "integratedTerminal"
        },          
        {
            "name": "Frontend (npm run dev)",
            "type": "node",
            "request": "launch",
            "runtimeExecutable": "npm",
            "runtimeArgs": ["run", "dev"],
            "cwd": "${workspaceFolder}/web",
            "console": "integratedTerminal",
            "sourceMaps": true,
            "outFiles": [
                "${workspaceFolder}/web/dist/**/*.js"
            ]
        }
    ],
    "compounds": [
        {
            "name": "Debug Backend and Frontend",
            "configurations": [
                "Debug Backend (launch_backend_service.sh)",
                "Frontend (npm run dev)"
            ]
        }
    ]
}
Could you help me resolve this issue? Thank you in advance for any guidance.

**Comment by peak-xu** [comment]
I want to know too
